src/client/src/consumer.rs

```diff
@@ -426,19 +426,25 @@ impl PartitionConsumer {
             .lookup_version(DefaultStreamFetchRequest::API_KEY)
             .unwrap_or((WASM_MODULE_API - 1) as i16);
 
-        if let Some(module) = config.wasm_module {
+        if let Some(mut module) = config.wasm_module {
             if stream_fetch_version < WASM_MODULE_API as i16 {
                 return Err(FluvioError::Other("SPU does not support WASM".to_owned()));
             }
 
             if stream_fetch_version < WASM_MODULE_V2_API as i16 {
                 // SmartStream V1
                 debug!("Using WASM V1 API");
-                let SmartStreamWasm::Raw(wasm) = module.wasm;
+                let wasm = module.wasm.get_raw()?;
                 stream_request.wasm_module = wasm;
             } else {
                 // SmartStream V2
                 debug!("Using WASM V2 API");
+                if stream_fetch_version < GZIP_WASM_API as i16 {
+                    module.wasm = module.wasm.to_raw()?;
```

Comment:
So here we're consuming `module.wasm` and just re-assigning a new value to itself. I think it would make more sense to use an `&mut self` pattern here for `to_raw` and `to_gzip`. That way, we can have

```rust
module.wasm.to_raw()?;
// instead of
module.wasm = module.wasm.to_raw()?;
```